src/Microsoft.VisualStudio.Threading/JoinableTask.cs

```diff
@@ -248,7 +248,9 @@ public Task Task
                     }
                 }
 
-                return this.wrappedTask as Task ?? this.GetTaskFromCompletionSource(this.wrappedTask);
+                // Read 'wrappedTask' once, so we know it's value is either a Task or a TaskCompletionSource.
```

Comment:
It took me a while to figure out why this change is important. Changing this code comment to the following may help make it more clear to others:

```suggestion
                // Guard against races in reading this field outside a lock while another thread writes to it
                // that would result in a `Task` object being passed to `GetTaskFromCompletionSource`
                // and a later InvalidCastException, by reading the field only once into a local variable.
```

Reply:
➡️ Comment rewritten